Makefile

```diff
@@ -1,63 +1,85 @@
 export GO15VENDOREXPERIMENT=1
 
-SHORT_NAME := deis-e2e
+SHORT_NAME := workflow-e2e
 
 SRC_PATH := /go/src/github.com/deis/workflow-e2e
-DEV_IMG := quay.io/deis/go-dev:0.9.1
-
-RUN_CMD := docker run --rm -e DEIS_ROUTER_SERVICE_HOST=${DEIS_ROUTER_SERVICE_HOST} -e DEIS_ROUTER_SERVICE_PORT=${DEIS_ROUTER_SERVICE_PORT} -v ${CURDIR}:${SRC_PATH} -w ${SRC_PATH} ${DEV_IMG}
-DEV_CMD := docker run --rm -e GO15VENDOREXPERIMENT=1 -v ${CURDIR}:${SRC_PATH} -w ${SRC_PATH} ${DEV_IMG}
-
-TEST_OPTS := -slowSpecThreshold=120.00 -noisyPendings=false
-PARALLEL_TEST_OPTS := ${TEST_OPTS} -p
 
 MUTABLE_VERSION ?= canary
 VERSION ?= git-$(shell git rev-parse --short HEAD)
 
+ifdef GINKGO_NODES
+export GINKO_NODES_ARG=-nodes=${GINKGO_NODES}
+else
+export GINKO_NODES_ARG=-p
+endif
+
+TEST_OPTS := -slowSpecThreshold=120.00 -noisyPendings=false ${GINKO_NODES_ARG}
+
 DEIS_REGISTRY ?= quay.io/
 IMAGE_PREFIX ?= deis
 IMAGE := ${DEIS_REGISTRY}${IMAGE_PREFIX}/${SHORT_NAME}:${VERSION}
 MUTABLE_IMAGE := ${DEIS_REGISTRY}${IMAGE_PREFIX}/${SHORT_NAME}:${MUTABLE_VERSION}
 
-.PHONY: bootstrap
+ifndef DEIS_CONTROLLER_URL
+ifdef DEIS_ROUTER_SERVICE_HOST
+export DEIS_CONTROLLER_URL=http://deis.${DEIS_ROUTER_SERVICE_HOST}.nip.io
+endif
+endif
```

Comment:
@arschles as we discussed offline, I've attempted this and there's a problem that arises from Ginkgo's architecture.  When executing tests in parallel, each Ginkgo node runs as its own process.  If the `DEIS_CONTROLLER_URL` environment variable is not set and is also not derivable from `DEIS_ROUTER_SERVICE_HOST`, you are warned about this once _per node_... and there isn't anything immediately obvious we can do about that.

We decided together to defer fixing this until beta3.

```
[kent@mbp workflow-e2e]$ make test-integration
ginkgo -slowSpecThreshold=120.00 -noisyPendings=false -nodes=15 tests/

	 -------------------------------------------------------------------
	|                                                                   |
	|  Ginkgo timed out waiting for all parallel nodes to report back!  |
	|                                                                   |
	 -------------------------------------------------------------------

[1] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[2] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[3] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[4] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[5] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[6] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[7] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[8] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[9] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[10] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[11] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[12] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[13] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[14] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!
[15] 2016/04/12 15:59:19 Environment variable DEIS_CONTROLLER_URL must be exported to proceed!

Ginkgo ran 1 suite in 3.560606126s
Test Suite Failed
make: *** [test-integration] Error 1
[kent@mbp workflow-e2e]$
```